### PR TITLE
[FEATURE] Generic Dialog Components

### PR DESCRIPTION
**Pull request by eunicorn**
https://github.com/perses/perses/discussions/935

Based on the discussion, we want generic dialog components that include common UX patterns:

1. Close icon button at the top right corner
2. Dividers for dialog content
3. For dialog actions, we always want primary button with contained variant on the left and secondary button (usually Cancel) with outlined variant on the right

### CODE REVIEW

File: ui/components/src/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { combineSx } from '../utils';
+
+export interface DialogHeaderProps extends DialogTitleProps {
+  /**
+   * Callback fired when close button is clicked. If undefined, close button will not appear in header.
+   */
+  onClose?: (e: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface DialogButtonProps {
```
Comment (celian-garcia) [inline ui/components/src/Dialog/Dialog.tsx:35]: You should probably extend the material ButtonProps here as well, to have all the power of a regular button. 

If you don't want the ``variant`` and ``type`` properties  you can use Omit

```ts
/**
 * following properties are removed from the DialogButtonProps
 */
interface UnwantedButtonProps {
  variant: unknown;
  type: unknown;
}

export interface DialogButtonProps extends Omit<ButtonProps, keyof UnwantedButtonProps> {}
```

File: ui/components/src/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+const Header = ({ children, onClose, ...props }: DialogHeaderProps) => {
+  return (
+    <>
+      <DialogTitle {...props}>{children}</DialogTitle>
+      {onClose && (
+        <IconButton
+          aria-label="Close"
+          onClick={onClose}
+          sx={(theme) => ({
```
Comment (celian-garcia) [inline ui/components/src/Dialog/Dialog.tsx:60]: In the Project CRUD PR, I wrapped this code in a function to optimize rendering performance (an advice from sonarlint). 

I'm not sure it's optimizing that much performances, but I found it clearer at the end. You can go and see if you find it interesting https://github.com/perses/perses/pull/916/files#diff-ce2596b6d1065537010136370c2489af315269aaafa78564f55f9bfd19cd80eeR20-R26

**Review by celian-garcia — CHANGES_REQUESTED** [review]
Great! That would remove already the main duplication issues in the dialogs.

File: ui/components/src/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+);
+
+const SecondaryButton = ({ children, onClick }: DialogButtonProps) => (
+  <Button variant="outlined" color="secondary" onClick={onClick}>
+    {children}
+  </Button>
+);
+
+export const Dialog = {
+  Dialog: MuiDialog,
```
Comment (saminzadeh) [inline ui/components/src/Dialog/Dialog.tsx:92]: `Dialog.Dialog` feels bit weird. A few ideas:

```
Dialog.Wrapper
Dialog.Base
```
or just expose this as `Dialog` and only add the sub components.
```tsx
export function Dialog() {...}
Dialog.Header = Header;
```

```tsx
<Dialog>
    <Dialog.Header>My Title</Dialog.Header>
</Dialog>

File: ui/components/src/Dialog/Dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import { combineSx } from '../utils';
+
+export interface DialogHeaderProps extends DialogTitleProps {
+  /**
+   * Callback fired when close button is clicked. If undefined, close button will not appear in header.
+   */
+  onClose?: (e: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface DialogButtonProps extends Omit<ButtonProps, 'variant' | 'color'> {
```
Comment (celian-garcia) [inline ui/components/src/Dialog/Dialog.tsx:38]: If you don't put ``type`` in the excluded properties, somebody using it will have some issues no ?

**Review by celian-garcia — APPROVED** [review]

**Comment by eunicorn** [comment]
@saminzadeh @cndonovan can one of you take another look when you have the chance? thanks!

**Review by saminzadeh — APPROVED** [review]

**Pull request merged**